pasion_preciouskaye/script.js

```diff
@@ -59,3 +59,60 @@ document.addEventListener('DOMContentLoaded', function () {
     });
   }
 });
+
+function searchCountry() {
+  var countryName = document.getElementById('countryInput').value.trim();
+  if (!countryName) {
+    document.getElementById('countryDetails').innerHTML = '<p>Please enter a country name.</p>';
+    document.getElementById('sameRegionCountries').innerHTML = '';
+    return;
+  }
+
+  fetch('https://restcountries.com/v3.1/name/' + countryName)
+    .then(function(response) {
+      if (!response.ok) {
+        throw new Error('Country not found');
+      }
+      return response.json();
+    })
+    .then(function(countryData) {
+      var country = countryData[0];
+      var details = `
+        <h2>Country Details - ${country.name.common}</h2>
+        <img src="${country.flags.svg}" alt="Flag of ${country.name.common}" width="100">
+        <p><strong>Area:</strong> ${country.area ? country.area.toLocaleString() + ' square kilometers' : 'N/A'}</p>
+        <p><strong>Languages:</strong> ${country.languages ? Object.values(country.languages).join(', ') : 'N/A'}</p>
+        <p><strong>Subregion:</strong> ${country.subregion ? country.subregion : 'N/A'}</p>
+        <p><strong>Capital:</strong> ${country.capital ? country.capital[0] : 'N/A'}</p>
+        <p><strong>Timezones:</strong> ${country.timezones ? country.timezones.join(', ') : 'N/A'}</p>`;
+      document.getElementById('countryDetails').innerHTML = details;
+
+      return fetch('https://restcountries.com/v3.1/region/' + country.region);
+    })
+    .then(function(response) {
+      if (!response.ok) {
+        throw new Error('Region not found');
+      }
+      return response.json();
+    })
+    .then(function(regionData) {
+      var region = regionData[0].region;
+      var sameRegionCountriesList = regionData.map(function(c) {
+        return `
+          <div style="display: inline-block; margin: 10px; text-align: center;">
+            <img src="${c.flags.svg}" alt="Flag of ${c.name.common}" width="50">
```

Comment:
Styling should be done in CSS. These lines also exceed 80 col.

References:
https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit?fbclid=IwZXh0bgNhZW0CMTAAAR2pLh7HJcCih-Pc8Oaz6b_cP15loflENQbWSdY9znNOhVtz6btuuPor-mg_aem_AeY-yfaZBMl4190ogpxCathoC0JDCMihSZ10oSLOVh2P_-Iz7Kne8nVbdVxfU2DPvVyYIvRs6hoKQNKdDUEIlvvY#slide=id.g25d183194fb_0_0

https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit?fbclid=IwZXh0bgNhZW0CMTAAAR2pLh7HJcCih-Pc8Oaz6b_cP15loflENQbWSdY9znNOhVtz6btuuPor-mg_aem_AeY-yfaZBMl4190ogpxCathoC0JDCMihSZ10oSLOVh2P_-Iz7Kne8nVbdVxfU2DPvVyYIvRs6hoKQNKdDUEIlvvY#slide=id.g29e6ce31160_0_11